### PR TITLE
[com_actionlogs] - add indexes for perfomance

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #22625 .

### Summary of Changes

added indexes

### Testing Instructions
see #22625

### CODE REVIEW

**Comment by PhilETaylor** [comment]
Please also add

`ALTER TABLE `#__action_logs` ADD INDEX (`extension`, `item_id`);`

**Comment by alikon** [comment]
i'll consider it too as well,....
in the meantime can you share some metrics with and without the indexes...

**Comment by PhilETaylor** [comment]
> in the meantime can you share some metrics with and without the indexes...

I did here https://github.com/joomla/joomla-cms/issues/22625#issuecomment-429762699

There is no point in even looking at metrics - the table had ZERO optimisation at all !

File: installation/sql/postgresql/joomla.sql [hunk 1]
```diff
@@ -2144,6 +2144,11 @@ CREATE TABLE "#__action_logs" (
   PRIMARY KEY ("id")
 );
 
+CREATE INDEX "#__action_logs_idx_user_id" ON "#__action_logs" ("user_id"); 
+CREATE INDEX "#__action_logs_idx_user_id_logdate" ON "#__action_logs" ("user_id", "log_date"); 
+CREATE INDEX "#__action_logs_idx_user_id_extension ON "#__action_logs" ("user_id", "extension");
```
Comment (Quy) [inline installation/sql/postgresql/joomla.sql]: Missing `"` before `ON`

Comment (alikon) [reply installation/sql/postgresql/joomla.sql]: fixed thanks

File: installation/sql/mysql/joomla.sql [hunk 2]
```diff
@@ -2158,6 +2158,9 @@ CREATE TABLE IF NOT EXISTS `#__action_logs` (
   `item_id` int(11) NOT NULL DEFAULT 0,
   `ip_address` VARCHAR(40) NOT NULL DEFAULT '0.0.0.0',
   PRIMARY KEY (`id`)
+  KEY (`user_id`),
+  KEY (`user_id`, `log_date`),
+  KEY (`user_id`, `extension`)
```
Comment (PhilETaylor) [inline installation/sql/mysql/joomla.sql]: need to add (`extension`, `item_id`)

Comment (alikon) [reply installation/sql/mysql/joomla.sql]: done

File: installation/sql/mysql/joomla.sql [hunk 3]
```diff
@@ -2158,6 +2158,10 @@ CREATE TABLE IF NOT EXISTS `#__action_logs` (
   `item_id` int(11) NOT NULL DEFAULT 0,
   `ip_address` VARCHAR(40) NOT NULL DEFAULT '0.0.0.0',
   PRIMARY KEY (`id`)
+  KEY (`user_id`),
+  KEY (`user_id`, `log_date`),
+  KEY (`user_id`, `extension`),
+	KEY (`extension`, `item_id`)
```
Comment (Quy) [inline installation/sql/mysql/joomla.sql]: Align `KEY`

Comment (alikon) [reply installation/sql/mysql/joomla.sql]: done

File: installation/sql/postgresql/joomla.sql [hunk 4]
```diff
@@ -2144,6 +2144,12 @@ CREATE TABLE "#__action_logs" (
   PRIMARY KEY ("id")
 );
 
+CREATE INDEX "#__action_logs_idx_user_id" ON "#__action_logs" ("user_id"); 
+CREATE INDEX "#__action_logs_idx_user_id_logdate" ON "#__action_logs" ("user_id", "log_date"); 
+CREATE INDEX "#__action_logs_idx_user_id_extension" ON "#__action_logs" ("user_id", "extension");
+CREATE INDEX "#__action_logs_idx_extension_itemid" ON "#__action_logs" ("extension", "item_id");
+
```
Comment (Quy) [inline installation/sql/postgresql/joomla.sql]: Remove tabs. Repeat in next line.

**Review by mbabker — APPROVED** [review]

**Pull request merged**